conf/index.html

```diff
@@ -114,32 +149,115 @@
+                this.pingInterval = setInterval(() => {
+                    this.ping();
+                }, 30000);
+            }
+
+            stopPing() {
+                if (this.pingInterval) {
+                    clearInterval(this.pingInterval);
+                    this.pingInterval = null;
+                }
+            }
+
+            scheduleRetry() {
+                this.retryCount++;
+                const currentInterval = this.retryIntervals[Math.min(this.retryCount - 1, this.retryIntervals.length - 1)];
+                const seconds = currentInterval / 1000;
+                setStatusMessage(`Connection lost. Retrying in ${seconds} seconds... (${this.retryCount}/${this.maxRetries})`);
+                setLoaderState('loading');
+                setShowLoader(true);
+                
+                this.retryTimeout = setTimeout(() => {
+                    console.log(`Attempting reconnection ${this.retryCount}/${this.maxRetries}`);
+                    this.connect();
+                }, currentInterval);
+            }
+
+            cleanup() {
+                this.stopPing();
+                if (this.retryTimeout) {
+                    clearTimeout(this.retryTimeout);
+                    this.retryTimeout = null;
+                }
+            }
+
+            async ping() {
                 try {
+                    await this.call('system_info');
+                } catch (error) {
+                    console.error('Ping failed:', error);
+                }
+            }
+
+            async onConnected() {
+                 try {
                     let accessKey = ''
-                    setStatusMessage('Generating TNC Registration URL');
+                    setStatusMessage('Getting things ready...');
                     const isAdopted = await this.call('is_adopted');
                     if (!isAdopted) {
                         accessKey = await this.call('adopt');
                         localStorage.setItem('accessKey', accessKey);
                     } else {
                         accessKey = localStorage.getItem('accessKey');
                     }
+                    if (!accessKey) {
+                        throw({ message : 'No Access Key'})
+                    }
 
                     await this.call('authenticate', accessKey)
+                    let systemInfo = await this.call('system_info')
+                    if (systemInfo.installation_running) {
+                        setStatusMessage('Installation in progress');
+                        return;
+                    }
                     let tncConfig = await this.call('tnc_config')
+                    if (tncConfig.initialization_error) {
+                         throw({ message : tncConfig.initialization_error})
+                    }
+                    if (Object.keys(tncConfig.registration_details).length) {
+                        setStatusMessage('Please continue the setup in TNC UI');
```

Comment:
Have we already discussed "TNC UI" as a representative of Connect?
TNC UI seems like a slang for employees to me. Have we considered simplifying to Connect Portal or something?

@chrismillsapsix @aervin